dotnet/src/webdriver/PinnedScript.cs

```diff
@@ -17,83 +17,70 @@
 // under the License.
 // </copyright>
 
-using System;
 using System.Globalization;
 
+#nullable enable
+
 namespace OpenQA.Selenium
 {
     /// <summary>
     /// A class representing a pinned JavaScript function that can be repeatedly called
     /// without sending the entire script across the wire for every execution.
     /// </summary>
-    public class PinnedScript
+    public sealed class PinnedScript
     {
-        private string scriptSource;
-        private string scriptHandle;
-        private string scriptId;
-
         /// <summary>
         /// Initializes a new instance of the <see cref="PinnedScript"/> class.
         /// </summary>
         /// <param name="script">The body of the JavaScript function to pin.</param>
+        /// <param name="stringHandle">The unique handle for this pinned script.</param>
+        /// <param name="scriptId">The internal ID of this script.</param>
         /// <remarks>
         /// This constructor is explicitly internal. Creation of pinned script objects
         /// is strictly the perview of Selenium, and should not be required by external
         /// libraries.
         /// </remarks>
-        internal PinnedScript(string script)
+        internal PinnedScript(string script, string stringHandle, string scriptId)
         {
-            this.scriptSource = script;
```

Comment:
It was fields, you propose properties. Keep it as field? Or it is used somewhere else outside?

Reply:
Users can access this value my calling the `myScript.Source` property. We populate it once in the constructor, and never change the value again. This is the perfect scenario for using an auto-property

Reply:
I am stupid, left comment for incorrect line.